// browserstack-node-sdk/generated/sdk-messages-accessibility_pb.js
// source: sdk-messages-accessibility.proto
/**
 * @fileoverview
 * @enhanceable
 * @suppress {missingRequire} reports error on implicit type usages.
 * @suppress {messageConventions} JS Compiler reports an error if a variable or
 *     field starts with 'MSG_' and isn't a translatable message.
 * @public
 */
// GENERATED CODE -- DO NOT EDIT!
/* eslint-disable */
// @ts-nocheck

var jspb = require('google-protobuf');
var goog = jspb;
var global = (function() {
  if (this) { return this; }
  if (typeof window !== 'undefined') { return window; }
  if (typeof global !== 'undefined') { return global; }
  if (typeof self !== 'undefined') { return self; }
  return Function('return this')();
}.call(null));

var sdk$messages$observability_pb = require('./sdk-messages-observability_pb.js');
goog.object.extend(proto, sdk$messages$observability_pb);
var sdk$messages$testhub_pb = require('./sdk-messages-testhub_pb.js');
goog.object.extend(proto, sdk$messages$testhub_pb);
goog.exportSymbol('proto.browserstack.sdk.v1.Accessibility', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.AccessibilityConfigRequest', null, global);
goog.exportSymbol('proto.browserstack.sdk.v1.AccessibilityConfigResponse', null, global);
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.Accessibility = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.browserstack.sdk.v1.Accessibility.repeatedFields_, null);
};
goog.inherits(proto.browserstack.sdk.v1.Accessibility, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.Accessibility.displayName = 'proto.browserstack.sdk.v1.Accessibility';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.repeatedFields_, null);
};
goog.inherits(proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.displayName = 'proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.repeatedFields_, null);
};
goog.inherits(proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.displayName = 'proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.displayName = 'proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.displayName = 'proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.displayName = 'proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.browserstack.sdk.v1.AccessibilityConfigRequest, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.AccessibilityConfigRequest.displayName = 'proto.browserstack.sdk.v1.AccessibilityConfigRequest';
}
/**
 * Generated by JsPbCodeGenerator.
 * @param {Array=} opt_data Optional initial data array, typically from a
 * server response, or constructed directly in Javascript. The array is used
 * in place and becomes part of the constructed object. It is not cloned.
 * If no data is provided, the constructed object will be empty, but still
 * valid.
 * @extends {jspb.Message}
 * @constructor
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse = function(opt_data) {
  jspb.Message.initialize(this, opt_data, 0, -1, null, null);
};
goog.inherits(proto.browserstack.sdk.v1.AccessibilityConfigResponse, jspb.Message);
if (goog.DEBUG && !COMPILED) {
  /**
   * @public
   * @override
   */
  proto.browserstack.sdk.v1.AccessibilityConfigResponse.displayName = 'proto.browserstack.sdk.v1.AccessibilityConfigResponse';
}

/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.browserstack.sdk.v1.Accessibility.repeatedFields_ = [5];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.Accessibility.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.Accessibility} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.toObject = function(includeInstance, msg) {
  var f, obj = {
    success: jspb.Message.getBooleanFieldWithDefault(msg, 1, false),
    options: (f = msg.getOptions()) && proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.toObject(includeInstance, f),
    errorTraceId: jspb.Message.getFieldWithDefault(msg, 3, ""),
    code: jspb.Message.getFieldWithDefault(msg, 4, ""),
    errorsList: jspb.Message.toObjectList(msg.getErrorsList(),
    sdk$messages$observability_pb.KeyMessage.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.Accessibility}
 */
proto.browserstack.sdk.v1.Accessibility.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.Accessibility;
  return proto.browserstack.sdk.v1.Accessibility.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.Accessibility} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.Accessibility}
 */
proto.browserstack.sdk.v1.Accessibility.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setSuccess(value);
      break;
    case 2:
      var value = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions;
      reader.readMessage(value,proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.deserializeBinaryFromReader);
      msg.setOptions(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setErrorTraceId(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setCode(value);
      break;
    case 5:
      var value = new sdk$messages$observability_pb.KeyMessage;
      reader.readMessage(value,sdk$messages$observability_pb.KeyMessage.deserializeBinaryFromReader);
      msg.addErrors(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.Accessibility.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.Accessibility} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSuccess();
  if (f) {
    writer.writeBool(
      1,
      f
    );
  }
  f = message.getOptions();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.serializeBinaryToWriter
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 3));
  if (f != null) {
    writer.writeString(
      3,
      f
    );
  }
  f = /** @type {string} */ (jspb.Message.getField(message, 4));
  if (f != null) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getErrorsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      5,
      f,
      sdk$messages$observability_pb.KeyMessage.serializeBinaryToWriter
    );
  }
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.repeatedFields_ = [2,3];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.toObject = function(includeInstance, msg) {
  var f, obj = {
    status: jspb.Message.getFieldWithDefault(msg, 1, ""),
    capabilitiesList: jspb.Message.toObjectList(msg.getCapabilitiesList(),
    proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.toObject, includeInstance),
    scriptsList: jspb.Message.toObjectList(msg.getScriptsList(),
    proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.toObject, includeInstance),
    commandsToWrap: (f = msg.getCommandsToWrap()) && proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions;
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setStatus(value);
      break;
    case 2:
      var value = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability;
      reader.readMessage(value,proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.deserializeBinaryFromReader);
      msg.addCapabilities(value);
      break;
    case 3:
      var value = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript;
      reader.readMessage(value,proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.deserializeBinaryFromReader);
      msg.addScripts(value);
      break;
    case 4:
      var value = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap;
      reader.readMessage(value,proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.deserializeBinaryFromReader);
      msg.setCommandsToWrap(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = /** @type {string} */ (jspb.Message.getField(message, 1));
  if (f != null) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCapabilitiesList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.serializeBinaryToWriter
    );
  }
  f = message.getScriptsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      3,
      f,
      proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.serializeBinaryToWriter
    );
  }
  f = message.getCommandsToWrap();
  if (f != null) {
    writer.writeMessage(
      4,
      f,
      proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.serializeBinaryToWriter
    );
  }
};



/**
 * List of repeated fields within this message type.
 * @private {!Array<number>}
 * @const
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.repeatedFields_ = [1,2];



if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.toObject = function(includeInstance, msg) {
  var f, obj = {
    scriptsToRunList: (f = jspb.Message.getRepeatedField(msg, 1)) == null ? undefined : f,
    commandsList: jspb.Message.toObjectList(msg.getCommandsList(),
    proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.toObject, includeInstance)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap;
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.addScriptsToRun(value);
      break;
    case 2:
      var value = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand;
      reader.readMessage(value,proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.deserializeBinaryFromReader);
      msg.addCommands(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getScriptsToRunList();
  if (f.length > 0) {
    writer.writeRepeatedString(
      1,
      f
    );
  }
  f = message.getCommandsList();
  if (f.length > 0) {
    writer.writeRepeatedMessage(
      2,
      f,
      proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.serializeBinaryToWriter
    );
  }
};


/**
 * repeated string scripts_to_run = 1;
 * @return {!Array<string>}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.getScriptsToRunList = function() {
  return /** @type {!Array<string>} */ (jspb.Message.getRepeatedField(this, 1));
};


/**
 * @param {!Array<string>} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.setScriptsToRunList = function(value) {
  return jspb.Message.setField(this, 1, value || []);
};


/**
 * @param {string} value
 * @param {number=} opt_index
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.addScriptsToRun = function(value, opt_index) {
  return jspb.Message.addToRepeatedField(this, 1, value, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.clearScriptsToRunList = function() {
  return this.setScriptsToRunList([]);
};


/**
 * repeated AccessibilityCommand commands = 2;
 * @return {!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand>}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.getCommandsList = function() {
  return /** @type{!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand, 2));
};


/**
 * @param {!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand>} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} returns this
*/
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.setCommandsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand=} opt_value
 * @param {number=} opt_index
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.addCommands = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap.prototype.clearCommandsList = function() {
  return this.setCommandsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    library: jspb.Message.getFieldWithDefault(msg, 2, ""),
    module: jspb.Message.getFieldWithDefault(msg, 3, ""),
    pb_class: jspb.Message.getFieldWithDefault(msg, 4, ""),
    method: jspb.Message.getFieldWithDefault(msg, 5, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand;
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setLibrary(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setModule(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setClass(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setMethod(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getLibrary();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
  f = message.getModule();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getClass();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getMethod();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string library = 2;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.getLibrary = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.setLibrary = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string module = 3;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.getModule = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.setModule = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string class = 4;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.getClass = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.setClass = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string method = 5;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.getMethod = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommand.prototype.setMethod = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    command: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript;
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setCommand(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getCommand();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string command = 2;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.prototype.getCommand = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript.prototype.setCommand = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.toObject = function(includeInstance, msg) {
  var f, obj = {
    name: jspb.Message.getFieldWithDefault(msg, 1, ""),
    value: jspb.Message.getFieldWithDefault(msg, 2, "")
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability;
  return proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setName(value);
      break;
    case 2:
      var value = /** @type {string} */ (reader.readString());
      msg.setValue(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getName();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getValue();
  if (f.length > 0) {
    writer.writeString(
      2,
      f
    );
  }
};


/**
 * optional string name = 1;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.prototype.getName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.prototype.setName = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional string value = 2;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.prototype.getValue = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 2, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability.prototype.setValue = function(value) {
  return jspb.Message.setProto3StringField(this, 2, value);
};


/**
 * optional string status = 1;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.getStatus = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.setStatus = function(value) {
  return jspb.Message.setField(this, 1, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.clearStatus = function() {
  return jspb.Message.setField(this, 1, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.hasStatus = function() {
  return jspb.Message.getField(this, 1) != null;
};


/**
 * repeated AccessibilityCapability capabilities = 2;
 * @return {!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability>}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.getCapabilitiesList = function() {
  return /** @type{!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability, 2));
};


/**
 * @param {!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability>} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
*/
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.setCapabilitiesList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 2, value);
};


/**
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability=} opt_value
 * @param {number=} opt_index
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.addCapabilities = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 2, opt_value, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCapability, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.clearCapabilitiesList = function() {
  return this.setCapabilitiesList([]);
};


/**
 * repeated AccessibilityScript scripts = 3;
 * @return {!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript>}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.getScriptsList = function() {
  return /** @type{!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript>} */ (
    jspb.Message.getRepeatedWrapperField(this, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript, 3));
};


/**
 * @param {!Array<!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript>} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
*/
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.setScriptsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 3, value);
};


/**
 * @param {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript=} opt_value
 * @param {number=} opt_index
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.addScripts = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 3, opt_value, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityScript, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.clearScriptsList = function() {
  return this.setScriptsList([]);
};


/**
 * optional AccessibilityCommandsToWrap commands_to_wrap = 4;
 * @return {?proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.getCommandsToWrap = function() {
  return /** @type{?proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap} */ (
    jspb.Message.getWrapperField(this, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap, 4));
};


/**
 * @param {?proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.AccessibilityCommandsToWrap|undefined} value
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
*/
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.setCommandsToWrap = function(value) {
  return jspb.Message.setWrapperField(this, 4, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} returns this
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.clearCommandsToWrap = function() {
  return this.setCommandsToWrap(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions.prototype.hasCommandsToWrap = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * optional bool success = 1;
 * @return {boolean}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.getSuccess = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.setSuccess = function(value) {
  return jspb.Message.setProto3BooleanField(this, 1, value);
};


/**
 * optional AccessibilityOptions options = 2;
 * @return {?proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.getOptions = function() {
  return /** @type{?proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions} */ (
    jspb.Message.getWrapperField(this, proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions, 2));
};


/**
 * @param {?proto.browserstack.sdk.v1.Accessibility.AccessibilityOptions|undefined} value
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
*/
proto.browserstack.sdk.v1.Accessibility.prototype.setOptions = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.clearOptions = function() {
  return this.setOptions(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.hasOptions = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional string error_trace_id = 3;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.getErrorTraceId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.setErrorTraceId = function(value) {
  return jspb.Message.setField(this, 3, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.clearErrorTraceId = function() {
  return jspb.Message.setField(this, 3, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.hasErrorTraceId = function() {
  return jspb.Message.getField(this, 3) != null;
};


/**
 * optional string code = 4;
 * @return {string}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.getCode = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.setCode = function(value) {
  return jspb.Message.setField(this, 4, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.clearCode = function() {
  return jspb.Message.setField(this, 4, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.hasCode = function() {
  return jspb.Message.getField(this, 4) != null;
};


/**
 * repeated KeyMessage errors = 5;
 * @return {!Array<!proto.browserstack.sdk.v1.KeyMessage>}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.getErrorsList = function() {
  return /** @type{!Array<!proto.browserstack.sdk.v1.KeyMessage>} */ (
    jspb.Message.getRepeatedWrapperField(this, sdk$messages$observability_pb.KeyMessage, 5));
};


/**
 * @param {!Array<!proto.browserstack.sdk.v1.KeyMessage>} value
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
*/
proto.browserstack.sdk.v1.Accessibility.prototype.setErrorsList = function(value) {
  return jspb.Message.setRepeatedWrapperField(this, 5, value);
};


/**
 * @param {!proto.browserstack.sdk.v1.KeyMessage=} opt_value
 * @param {number=} opt_index
 * @return {!proto.browserstack.sdk.v1.KeyMessage}
 */
proto.browserstack.sdk.v1.Accessibility.prototype.addErrors = function(opt_value, opt_index) {
  return jspb.Message.addToRepeatedWrapperField(this, 5, opt_value, proto.browserstack.sdk.v1.KeyMessage, opt_index);
};


/**
 * Clears the list making it empty but non-null.
 * @return {!proto.browserstack.sdk.v1.Accessibility} returns this
 */
proto.browserstack.sdk.v1.Accessibility.prototype.clearErrorsList = function() {
  return this.setErrorsList([]);
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.AccessibilityConfigRequest.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.toObject = function(includeInstance, msg) {
  var f, obj = {
    binSessionId: jspb.Message.getFieldWithDefault(msg, 1, ""),
    platformIndex: jspb.Message.getFieldWithDefault(msg, 2, 0),
    frameworkName: jspb.Message.getFieldWithDefault(msg, 3, ""),
    frameworkVersion: jspb.Message.getFieldWithDefault(msg, 4, ""),
    hubUrl: jspb.Message.getFieldWithDefault(msg, 5, ""),
    capabilities: msg.getCapabilities_asB64()
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.AccessibilityConfigRequest;
  return proto.browserstack.sdk.v1.AccessibilityConfigRequest.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {string} */ (reader.readString());
      msg.setBinSessionId(value);
      break;
    case 2:
      var value = /** @type {number} */ (reader.readUint32());
      msg.setPlatformIndex(value);
      break;
    case 3:
      var value = /** @type {string} */ (reader.readString());
      msg.setFrameworkName(value);
      break;
    case 4:
      var value = /** @type {string} */ (reader.readString());
      msg.setFrameworkVersion(value);
      break;
    case 5:
      var value = /** @type {string} */ (reader.readString());
      msg.setHubUrl(value);
      break;
    case 6:
      var value = /** @type {!Uint8Array} */ (reader.readBytes());
      msg.setCapabilities(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.AccessibilityConfigRequest.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getBinSessionId();
  if (f.length > 0) {
    writer.writeString(
      1,
      f
    );
  }
  f = message.getPlatformIndex();
  if (f !== 0) {
    writer.writeUint32(
      2,
      f
    );
  }
  f = message.getFrameworkName();
  if (f.length > 0) {
    writer.writeString(
      3,
      f
    );
  }
  f = message.getFrameworkVersion();
  if (f.length > 0) {
    writer.writeString(
      4,
      f
    );
  }
  f = message.getHubUrl();
  if (f.length > 0) {
    writer.writeString(
      5,
      f
    );
  }
  f = /** @type {!(string|Uint8Array)} */ (jspb.Message.getField(message, 6));
  if (f != null) {
    writer.writeBytes(
      6,
      f
    );
  }
};


/**
 * optional string bin_session_id = 1;
 * @return {string}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getBinSessionId = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 1, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.setBinSessionId = function(value) {
  return jspb.Message.setProto3StringField(this, 1, value);
};


/**
 * optional uint32 platform_index = 2;
 * @return {number}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getPlatformIndex = function() {
  return /** @type {number} */ (jspb.Message.getFieldWithDefault(this, 2, 0));
};


/**
 * @param {number} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.setPlatformIndex = function(value) {
  return jspb.Message.setProto3IntField(this, 2, value);
};


/**
 * optional string framework_name = 3;
 * @return {string}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getFrameworkName = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 3, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.setFrameworkName = function(value) {
  return jspb.Message.setProto3StringField(this, 3, value);
};


/**
 * optional string framework_version = 4;
 * @return {string}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getFrameworkVersion = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 4, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.setFrameworkVersion = function(value) {
  return jspb.Message.setProto3StringField(this, 4, value);
};


/**
 * optional string hub_url = 5;
 * @return {string}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getHubUrl = function() {
  return /** @type {string} */ (jspb.Message.getFieldWithDefault(this, 5, ""));
};


/**
 * @param {string} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.setHubUrl = function(value) {
  return jspb.Message.setProto3StringField(this, 5, value);
};


/**
 * optional bytes capabilities = 6;
 * @return {!(string|Uint8Array)}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getCapabilities = function() {
  return /** @type {!(string|Uint8Array)} */ (jspb.Message.getFieldWithDefault(this, 6, ""));
};


/**
 * optional bytes capabilities = 6;
 * This is a type-conversion wrapper around `getCapabilities()`
 * @return {string}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getCapabilities_asB64 = function() {
  return /** @type {string} */ (jspb.Message.bytesAsB64(
      this.getCapabilities()));
};


/**
 * optional bytes capabilities = 6;
 * Note that Uint8Array is not supported on all browsers.
 * @see http://caniuse.com/Uint8Array
 * This is a type-conversion wrapper around `getCapabilities()`
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.getCapabilities_asU8 = function() {
  return /** @type {!Uint8Array} */ (jspb.Message.bytesAsU8(
      this.getCapabilities()));
};


/**
 * @param {!(string|Uint8Array)} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.setCapabilities = function(value) {
  return jspb.Message.setField(this, 6, value);
};


/**
 * Clears the field making it undefined.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigRequest} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.clearCapabilities = function() {
  return jspb.Message.setField(this, 6, undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.AccessibilityConfigRequest.prototype.hasCapabilities = function() {
  return jspb.Message.getField(this, 6) != null;
};





if (jspb.Message.GENERATE_TO_OBJECT) {
/**
 * Creates an object representation of this proto.
 * Field names that are reserved in JavaScript and will be renamed to pb_name.
 * Optional fields that are not set will be set to undefined.
 * To access a reserved field use, foo.pb_<name>, eg, foo.pb_default.
 * For the list of reserved names please see:
 *     net/proto2/compiler/js/internal/generator.cc#kKeyword.
 * @param {boolean=} opt_includeInstance Deprecated. whether to include the
 *     JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @return {!Object}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.toObject = function(opt_includeInstance) {
  return proto.browserstack.sdk.v1.AccessibilityConfigResponse.toObject(opt_includeInstance, this);
};


/**
 * Static version of the {@see toObject} method.
 * @param {boolean|undefined} includeInstance Deprecated. Whether to include
 *     the JSPB instance for transitional soy proto support:
 *     http://goto/soy-param-migration
 * @param {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} msg The msg instance to transform.
 * @return {!Object}
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.toObject = function(includeInstance, msg) {
  var f, obj = {
    success: jspb.Message.getBooleanFieldWithDefault(msg, 1, false),
    testhub: (f = msg.getTesthub()) && sdk$messages$testhub_pb.TestHub.toObject(includeInstance, f),
    accessibility: (f = msg.getAccessibility()) && proto.browserstack.sdk.v1.Accessibility.toObject(includeInstance, f)
  };

  if (includeInstance) {
    obj.$jspbMessageInstance = msg;
  }
  return obj;
};
}


/**
 * Deserializes binary data (in protobuf wire format).
 * @param {jspb.ByteSource} bytes The bytes to deserialize.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.deserializeBinary = function(bytes) {
  var reader = new jspb.BinaryReader(bytes);
  var msg = new proto.browserstack.sdk.v1.AccessibilityConfigResponse;
  return proto.browserstack.sdk.v1.AccessibilityConfigResponse.deserializeBinaryFromReader(msg, reader);
};


/**
 * Deserializes binary data (in protobuf wire format) from the
 * given reader into the given message object.
 * @param {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} msg The message object to deserialize into.
 * @param {!jspb.BinaryReader} reader The BinaryReader to use.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.deserializeBinaryFromReader = function(msg, reader) {
  while (reader.nextField()) {
    if (reader.isEndGroup()) {
      break;
    }
    var field = reader.getFieldNumber();
    switch (field) {
    case 1:
      var value = /** @type {boolean} */ (reader.readBool());
      msg.setSuccess(value);
      break;
    case 2:
      var value = new sdk$messages$testhub_pb.TestHub;
      reader.readMessage(value,sdk$messages$testhub_pb.TestHub.deserializeBinaryFromReader);
      msg.setTesthub(value);
      break;
    case 3:
      var value = new proto.browserstack.sdk.v1.Accessibility;
      reader.readMessage(value,proto.browserstack.sdk.v1.Accessibility.deserializeBinaryFromReader);
      msg.setAccessibility(value);
      break;
    default:
      reader.skipField();
      break;
    }
  }
  return msg;
};


/**
 * Serializes the message to binary data (in protobuf wire format).
 * @return {!Uint8Array}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.serializeBinary = function() {
  var writer = new jspb.BinaryWriter();
  proto.browserstack.sdk.v1.AccessibilityConfigResponse.serializeBinaryToWriter(this, writer);
  return writer.getResultBuffer();
};


/**
 * Serializes the given message to binary data (in protobuf wire
 * format), writing to the given BinaryWriter.
 * @param {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} message
 * @param {!jspb.BinaryWriter} writer
 * @suppress {unusedLocalVariables} f is only used for nested messages
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.serializeBinaryToWriter = function(message, writer) {
  var f = undefined;
  f = message.getSuccess();
  if (f) {
    writer.writeBool(
      1,
      f
    );
  }
  f = message.getTesthub();
  if (f != null) {
    writer.writeMessage(
      2,
      f,
      sdk$messages$testhub_pb.TestHub.serializeBinaryToWriter
    );
  }
  f = message.getAccessibility();
  if (f != null) {
    writer.writeMessage(
      3,
      f,
      proto.browserstack.sdk.v1.Accessibility.serializeBinaryToWriter
    );
  }
};


/**
 * optional bool success = 1;
 * @return {boolean}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.getSuccess = function() {
  return /** @type {boolean} */ (jspb.Message.getBooleanFieldWithDefault(this, 1, false));
};


/**
 * @param {boolean} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.setSuccess = function(value) {
  return jspb.Message.setProto3BooleanField(this, 1, value);
};


/**
 * optional TestHub testhub = 2;
 * @return {?proto.browserstack.sdk.v1.TestHub}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.getTesthub = function() {
  return /** @type{?proto.browserstack.sdk.v1.TestHub} */ (
    jspb.Message.getWrapperField(this, sdk$messages$testhub_pb.TestHub, 2));
};


/**
 * @param {?proto.browserstack.sdk.v1.TestHub|undefined} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} returns this
*/
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.setTesthub = function(value) {
  return jspb.Message.setWrapperField(this, 2, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.clearTesthub = function() {
  return this.setTesthub(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.hasTesthub = function() {
  return jspb.Message.getField(this, 2) != null;
};


/**
 * optional Accessibility accessibility = 3;
 * @return {?proto.browserstack.sdk.v1.Accessibility}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.getAccessibility = function() {
  return /** @type{?proto.browserstack.sdk.v1.Accessibility} */ (
    jspb.Message.getWrapperField(this, proto.browserstack.sdk.v1.Accessibility, 3));
};


/**
 * @param {?proto.browserstack.sdk.v1.Accessibility|undefined} value
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} returns this
*/
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.setAccessibility = function(value) {
  return jspb.Message.setWrapperField(this, 3, value);
};


/**
 * Clears the message field making it undefined.
 * @return {!proto.browserstack.sdk.v1.AccessibilityConfigResponse} returns this
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.clearAccessibility = function() {
  return this.setAccessibility(undefined);
};


/**
 * Returns whether this field is set.
 * @return {boolean}
 */
proto.browserstack.sdk.v1.AccessibilityConfigResponse.prototype.hasAccessibility = function() {
  return jspb.Message.getField(this, 3) != null;
};


goog.object.extend(exports, proto.browserstack.sdk.v1);
