// browserstack-node-sdk/generated/sdk_grpc_pb.js
// GENERATED CODE -- DO NOT EDIT!

'use strict';
var grpc = require('@grpc/grpc-js');
var sdk$messages$ai_pb = require('./sdk-messages-ai_pb.js');
var sdk$messages$accessibility_pb = require('./sdk-messages-accessibility_pb.js');
var sdk$messages$observability_pb = require('./sdk-messages-observability_pb.js');
var sdk$messages_pb = require('./sdk-messages_pb.js');

function serialize_browserstack_sdk_v1_AIBrowserExtensionRequest(arg) {
  if (!(arg instanceof sdk$messages$ai_pb.AIBrowserExtensionRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AIBrowserExtensionRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AIBrowserExtensionRequest(buffer_arg) {
  return sdk$messages$ai_pb.AIBrowserExtensionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AIBrowserExtensionResponse(arg) {
  if (!(arg instanceof sdk$messages$ai_pb.AIBrowserExtensionResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AIBrowserExtensionResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AIBrowserExtensionResponse(buffer_arg) {
  return sdk$messages$ai_pb.AIBrowserExtensionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AISelfHealGetRequest(arg) {
  if (!(arg instanceof sdk$messages$ai_pb.AISelfHealGetRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AISelfHealGetRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AISelfHealGetRequest(buffer_arg) {
  return sdk$messages$ai_pb.AISelfHealGetRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AISelfHealGetResponse(arg) {
  if (!(arg instanceof sdk$messages$ai_pb.AISelfHealGetResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AISelfHealGetResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AISelfHealGetResponse(buffer_arg) {
  return sdk$messages$ai_pb.AISelfHealGetResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AISelfHealStepRequest(arg) {
  if (!(arg instanceof sdk$messages$ai_pb.AISelfHealStepRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AISelfHealStepRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AISelfHealStepRequest(buffer_arg) {
  return sdk$messages$ai_pb.AISelfHealStepRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AISelfHealStepResponse(arg) {
  if (!(arg instanceof sdk$messages$ai_pb.AISelfHealStepResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AISelfHealStepResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AISelfHealStepResponse(buffer_arg) {
  return sdk$messages$ai_pb.AISelfHealStepResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AccessibilityConfigRequest(arg) {
  if (!(arg instanceof sdk$messages$accessibility_pb.AccessibilityConfigRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AccessibilityConfigRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AccessibilityConfigRequest(buffer_arg) {
  return sdk$messages$accessibility_pb.AccessibilityConfigRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AccessibilityConfigResponse(arg) {
  if (!(arg instanceof sdk$messages$accessibility_pb.AccessibilityConfigResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AccessibilityConfigResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AccessibilityConfigResponse(buffer_arg) {
  return sdk$messages$accessibility_pb.AccessibilityConfigResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AutomationFrameworkInitRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.AutomationFrameworkInitRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AutomationFrameworkInitRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AutomationFrameworkInitRequest(buffer_arg) {
  return sdk$messages_pb.AutomationFrameworkInitRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AutomationFrameworkInitResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.AutomationFrameworkInitResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AutomationFrameworkInitResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AutomationFrameworkInitResponse(buffer_arg) {
  return sdk$messages_pb.AutomationFrameworkInitResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AutomationFrameworkStartRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.AutomationFrameworkStartRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AutomationFrameworkStartRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AutomationFrameworkStartRequest(buffer_arg) {
  return sdk$messages_pb.AutomationFrameworkStartRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AutomationFrameworkStartResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.AutomationFrameworkStartResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AutomationFrameworkStartResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AutomationFrameworkStartResponse(buffer_arg) {
  return sdk$messages_pb.AutomationFrameworkStartResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AutomationFrameworkStopRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.AutomationFrameworkStopRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AutomationFrameworkStopRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AutomationFrameworkStopRequest(buffer_arg) {
  return sdk$messages_pb.AutomationFrameworkStopRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_AutomationFrameworkStopResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.AutomationFrameworkStopResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.AutomationFrameworkStopResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_AutomationFrameworkStopResponse(buffer_arg) {
  return sdk$messages_pb.AutomationFrameworkStopResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_ConnectBinSessionRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.ConnectBinSessionRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.ConnectBinSessionRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_ConnectBinSessionRequest(buffer_arg) {
  return sdk$messages_pb.ConnectBinSessionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_ConnectBinSessionResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.ConnectBinSessionResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.ConnectBinSessionResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_ConnectBinSessionResponse(buffer_arg) {
  return sdk$messages_pb.ConnectBinSessionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_DriverInitRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.DriverInitRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.DriverInitRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_DriverInitRequest(buffer_arg) {
  return sdk$messages_pb.DriverInitRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_DriverInitResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.DriverInitResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.DriverInitResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_DriverInitResponse(buffer_arg) {
  return sdk$messages_pb.DriverInitResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_EnqueueTestEventRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.EnqueueTestEventRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.EnqueueTestEventRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_EnqueueTestEventRequest(buffer_arg) {
  return sdk$messages_pb.EnqueueTestEventRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_EnqueueTestEventResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.EnqueueTestEventResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.EnqueueTestEventResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_EnqueueTestEventResponse(buffer_arg) {
  return sdk$messages_pb.EnqueueTestEventResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_FindNearestHubRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.FindNearestHubRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.FindNearestHubRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_FindNearestHubRequest(buffer_arg) {
  return sdk$messages_pb.FindNearestHubRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_FindNearestHubResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.FindNearestHubResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.FindNearestHubResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_FindNearestHubResponse(buffer_arg) {
  return sdk$messages_pb.FindNearestHubResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_LogCreatedEventRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.LogCreatedEventRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.LogCreatedEventRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_LogCreatedEventRequest(buffer_arg) {
  return sdk$messages_pb.LogCreatedEventRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_LogCreatedEventResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.LogCreatedEventResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.LogCreatedEventResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_LogCreatedEventResponse(buffer_arg) {
  return sdk$messages_pb.LogCreatedEventResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_ObservabilityConfigRequest(arg) {
  if (!(arg instanceof sdk$messages$observability_pb.ObservabilityConfigRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.ObservabilityConfigRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_ObservabilityConfigRequest(buffer_arg) {
  return sdk$messages$observability_pb.ObservabilityConfigRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_ObservabilityConfigResponse(arg) {
  if (!(arg instanceof sdk$messages$observability_pb.ObservabilityConfigResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.ObservabilityConfigResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_ObservabilityConfigResponse(buffer_arg) {
  return sdk$messages$observability_pb.ObservabilityConfigResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_PlaywrightConfigRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.PlaywrightConfigRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.PlaywrightConfigRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_PlaywrightConfigRequest(buffer_arg) {
  return sdk$messages_pb.PlaywrightConfigRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_PlaywrightConfigResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.PlaywrightConfigResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.PlaywrightConfigResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_PlaywrightConfigResponse(buffer_arg) {
  return sdk$messages_pb.PlaywrightConfigResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_StartBinSessionRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.StartBinSessionRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.StartBinSessionRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_StartBinSessionRequest(buffer_arg) {
  return sdk$messages_pb.StartBinSessionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_StartBinSessionResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.StartBinSessionResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.StartBinSessionResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_StartBinSessionResponse(buffer_arg) {
  return sdk$messages_pb.StartBinSessionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_StopBinSessionRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.StopBinSessionRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.StopBinSessionRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_StopBinSessionRequest(buffer_arg) {
  return sdk$messages_pb.StopBinSessionRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_StopBinSessionResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.StopBinSessionResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.StopBinSessionResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_StopBinSessionResponse(buffer_arg) {
  return sdk$messages_pb.StopBinSessionResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_TestFrameworkEventRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.TestFrameworkEventRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.TestFrameworkEventRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_TestFrameworkEventRequest(buffer_arg) {
  return sdk$messages_pb.TestFrameworkEventRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_TestFrameworkEventResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.TestFrameworkEventResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.TestFrameworkEventResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_TestFrameworkEventResponse(buffer_arg) {
  return sdk$messages_pb.TestFrameworkEventResponse.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_TestSessionEventRequest(arg) {
  if (!(arg instanceof sdk$messages_pb.TestSessionEventRequest)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.TestSessionEventRequest');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_TestSessionEventRequest(buffer_arg) {
  return sdk$messages_pb.TestSessionEventRequest.deserializeBinary(new Uint8Array(buffer_arg));
}

function serialize_browserstack_sdk_v1_TestSessionEventResponse(arg) {
  if (!(arg instanceof sdk$messages_pb.TestSessionEventResponse)) {
    throw new Error('Expected argument of type browserstack.sdk.v1.TestSessionEventResponse');
  }
  return Buffer.from(arg.serializeBinary());
}

function deserialize_browserstack_sdk_v1_TestSessionEventResponse(buffer_arg) {
  return sdk$messages_pb.TestSessionEventResponse.deserializeBinary(new Uint8Array(buffer_arg));
}


var SDKService = exports.SDKService = {
  startBinSession: {
    path: '/browserstack.sdk.v1.SDK/StartBinSession',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.StartBinSessionRequest,
    responseType: sdk$messages_pb.StartBinSessionResponse,
    requestSerialize: serialize_browserstack_sdk_v1_StartBinSessionRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_StartBinSessionRequest,
    responseSerialize: serialize_browserstack_sdk_v1_StartBinSessionResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_StartBinSessionResponse,
  },
  playwrightConfig: {
    path: '/browserstack.sdk.v1.SDK/PlaywrightConfig',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.PlaywrightConfigRequest,
    responseType: sdk$messages_pb.PlaywrightConfigResponse,
    requestSerialize: serialize_browserstack_sdk_v1_PlaywrightConfigRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_PlaywrightConfigRequest,
    responseSerialize: serialize_browserstack_sdk_v1_PlaywrightConfigResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_PlaywrightConfigResponse,
  },
  connectBinSession: {
    path: '/browserstack.sdk.v1.SDK/ConnectBinSession',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.ConnectBinSessionRequest,
    responseType: sdk$messages_pb.ConnectBinSessionResponse,
    requestSerialize: serialize_browserstack_sdk_v1_ConnectBinSessionRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_ConnectBinSessionRequest,
    responseSerialize: serialize_browserstack_sdk_v1_ConnectBinSessionResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_ConnectBinSessionResponse,
  },
  stopBinSession: {
    path: '/browserstack.sdk.v1.SDK/StopBinSession',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.StopBinSessionRequest,
    responseType: sdk$messages_pb.StopBinSessionResponse,
    requestSerialize: serialize_browserstack_sdk_v1_StopBinSessionRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_StopBinSessionRequest,
    responseSerialize: serialize_browserstack_sdk_v1_StopBinSessionResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_StopBinSessionResponse,
  },
  driverInit: {
    path: '/browserstack.sdk.v1.SDK/DriverInit',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.DriverInitRequest,
    responseType: sdk$messages_pb.DriverInitResponse,
    requestSerialize: serialize_browserstack_sdk_v1_DriverInitRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_DriverInitRequest,
    responseSerialize: serialize_browserstack_sdk_v1_DriverInitResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_DriverInitResponse,
  },
  automationFrameworkInit: {
    path: '/browserstack.sdk.v1.SDK/AutomationFrameworkInit',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.AutomationFrameworkInitRequest,
    responseType: sdk$messages_pb.AutomationFrameworkInitResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AutomationFrameworkInitRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AutomationFrameworkInitRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AutomationFrameworkInitResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AutomationFrameworkInitResponse,
  },
  automationFrameworkStart: {
    path: '/browserstack.sdk.v1.SDK/AutomationFrameworkStart',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.AutomationFrameworkStartRequest,
    responseType: sdk$messages_pb.AutomationFrameworkStartResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AutomationFrameworkStartRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AutomationFrameworkStartRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AutomationFrameworkStartResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AutomationFrameworkStartResponse,
  },
  // session-start
automationFrameworkStop: {
    path: '/browserstack.sdk.v1.SDK/AutomationFrameworkStop',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.AutomationFrameworkStopRequest,
    responseType: sdk$messages_pb.AutomationFrameworkStopResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AutomationFrameworkStopRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AutomationFrameworkStopRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AutomationFrameworkStopResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AutomationFrameworkStopResponse,
  },
  findNearestHub: {
    path: '/browserstack.sdk.v1.SDK/FindNearestHub',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.FindNearestHubRequest,
    responseType: sdk$messages_pb.FindNearestHubResponse,
    requestSerialize: serialize_browserstack_sdk_v1_FindNearestHubRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_FindNearestHubRequest,
    responseSerialize: serialize_browserstack_sdk_v1_FindNearestHubResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_FindNearestHubResponse,
  },
  aIBrowserExtension: {
    path: '/browserstack.sdk.v1.SDK/AIBrowserExtension',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages$ai_pb.AIBrowserExtensionRequest,
    responseType: sdk$messages$ai_pb.AIBrowserExtensionResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AIBrowserExtensionRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AIBrowserExtensionRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AIBrowserExtensionResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AIBrowserExtensionResponse,
  },
  aISelfHealStep: {
    path: '/browserstack.sdk.v1.SDK/AISelfHealStep',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages$ai_pb.AISelfHealStepRequest,
    responseType: sdk$messages$ai_pb.AISelfHealStepResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AISelfHealStepRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AISelfHealStepRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AISelfHealStepResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AISelfHealStepResponse,
  },
  aISelfHealGetResult: {
    path: '/browserstack.sdk.v1.SDK/AISelfHealGetResult',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages$ai_pb.AISelfHealGetRequest,
    responseType: sdk$messages$ai_pb.AISelfHealGetResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AISelfHealGetRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AISelfHealGetRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AISelfHealGetResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AISelfHealGetResponse,
  },
  accessibilityConfig: {
    path: '/browserstack.sdk.v1.SDK/AccessibilityConfig',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages$accessibility_pb.AccessibilityConfigRequest,
    responseType: sdk$messages$accessibility_pb.AccessibilityConfigResponse,
    requestSerialize: serialize_browserstack_sdk_v1_AccessibilityConfigRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_AccessibilityConfigRequest,
    responseSerialize: serialize_browserstack_sdk_v1_AccessibilityConfigResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_AccessibilityConfigResponse,
  },
  observabilityConfig: {
    path: '/browserstack.sdk.v1.SDK/ObservabilityConfig',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages$observability_pb.ObservabilityConfigRequest,
    responseType: sdk$messages$observability_pb.ObservabilityConfigResponse,
    requestSerialize: serialize_browserstack_sdk_v1_ObservabilityConfigRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_ObservabilityConfigRequest,
    responseSerialize: serialize_browserstack_sdk_v1_ObservabilityConfigResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_ObservabilityConfigResponse,
  },
  testFrameworkEvent: {
    path: '/browserstack.sdk.v1.SDK/TestFrameworkEvent',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.TestFrameworkEventRequest,
    responseType: sdk$messages_pb.TestFrameworkEventResponse,
    requestSerialize: serialize_browserstack_sdk_v1_TestFrameworkEventRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_TestFrameworkEventRequest,
    responseSerialize: serialize_browserstack_sdk_v1_TestFrameworkEventResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_TestFrameworkEventResponse,
  },
  testSessionEvent: {
    path: '/browserstack.sdk.v1.SDK/TestSessionEvent',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.TestSessionEventRequest,
    responseType: sdk$messages_pb.TestSessionEventResponse,
    requestSerialize: serialize_browserstack_sdk_v1_TestSessionEventRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_TestSessionEventRequest,
    responseSerialize: serialize_browserstack_sdk_v1_TestSessionEventResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_TestSessionEventResponse,
  },
  enqueueTestEvent: {
    path: '/browserstack.sdk.v1.SDK/EnqueueTestEvent',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.EnqueueTestEventRequest,
    responseType: sdk$messages_pb.EnqueueTestEventResponse,
    requestSerialize: serialize_browserstack_sdk_v1_EnqueueTestEventRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_EnqueueTestEventRequest,
    responseSerialize: serialize_browserstack_sdk_v1_EnqueueTestEventResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_EnqueueTestEventResponse,
  },
  logCreatedEvent: {
    path: '/browserstack.sdk.v1.SDK/LogCreatedEvent',
    requestStream: false,
    responseStream: false,
    requestType: sdk$messages_pb.LogCreatedEventRequest,
    responseType: sdk$messages_pb.LogCreatedEventResponse,
    requestSerialize: serialize_browserstack_sdk_v1_LogCreatedEventRequest,
    requestDeserialize: deserialize_browserstack_sdk_v1_LogCreatedEventRequest,
    responseSerialize: serialize_browserstack_sdk_v1_LogCreatedEventResponse,
    responseDeserialize: deserialize_browserstack_sdk_v1_LogCreatedEventResponse,
  },
};

exports.SDKClient = grpc.makeGenericClientConstructor(SDKService, 'SDK');
